doc/decisions/6_implemented/gradle.md

```diff
@@ -0,0 +1,45 @@
+# Replacing Maven as Java-related build system with Gradle
+
+## Problem
+
+Originally, Maven was used as the build system for Elektra's Java-related modules. When it came time to modernize and improve Elekta's Java bindings, the question of which build system to use had to be answered. Ease of use, community support, and flexibility were the most important factors to consider. The fact that Maven does not provide native support for Kotlin, which is becoming increasingly popular with developers, was also an important factor to consider.
+
+## Constraints
+
+- Existing dependencies must be supported by the chosen build system
+- An established system with a supportive community would be of great advantage
+- Build performance should not be negatively impacted
+- Integration with existing build infrastructure is no more complicated than before
+- The benefits of the transition outweigh the potential costs, making it a worthwhile investment
+
+## Assumptions
+
+The development team either has the necessary expertise and resources for a successful transition, including knowledge of the new build system and familiarity with its syntax and conventions, or the new build system should be easy and intuitive for the development team to use, especially if the team is not already familiar with the new system.
+
+## Considered Alternatives
+
+- Ant is a popular build tool that uses XML files to define build processes. It is flexible and can be used for a variety of languages, but it can be more verbose and less user-friendly than other build tools.
+
+- Make is a build tool that uses make files to define build processes. It is popular for C/C++ projects and can be extensively customized, but it can be more difficult to use than other build tools and is not as flexible for projects that use multiple languages. Since Elektra is a C-based software project, this system would have had the advantage of already being in use.
```

Comment:
```suggestion
- CMake is a build tool, which only has limited support for Java.
  It is popular for C/C++ projects and can be extensively customized, but it can be more difficult to use than other build tools and is not as flexible for projects that use multiple languages.
  Since Elektra is a C-based software project, this system would have had the advantage of already being in use.
```